src/tests/ftest/util/soak_utils.py

```diff
@@ -522,66 +526,80 @@ def launch_reboot(self, pools, name, results, args):
-            status = False
-        if not wait_for_result(self.log, check_ssh, 120, 2, True, hosts=reboot_host,
-                               cmd_timeout=30, verbose=True):
-            self.log.error(f"<<<FAILED:{name} - {reboot_host} failed to reboot")
-            status = False
-        if status:
-            # issue a restart
-            self.log.info("<<<PASS %s: Issue systemctl restart daos_server on %s at %s>>>\n",
-                          self.loop, name, reboot_host, time.ctime())
-            cmd_results = run_remote(self.log, reboot_host, "sudo systemctl restart daos_server")
-            if cmd_results.passed:
-                self.dmg_command.system_query()
-                for pool in pools:
-                    self.dmg_command.pool_query(pool.identifier)
-                # wait server to be started
-                try:
-                    self.dmg_command.system_start(ranks=ranks)
-                except CommandFailure as error:
-                    self.log.error("<<<FAILED:dmg system start failed", exc_info=error)
-                    status = False
-                for pool in pools:
-                    self.dmg_command.pool_query(pool.identifier)
-                self.dmg_command.system_query()
-            else:
-                self.log.error("<<<FAILED:systemctl start daos_server failed")
+    elif name == "REBOOT_REINTEGRATE":
+        if self.harasser_results["REBOOT"]:
+            reboot_host = self.harasser_args["REBOOT"]["host"]
+            ranklist = self.harasser_args["REBOOT"]["ranks"]
+            ranks = ",".join(str(rank) for rank in ranklist)
+            self.log.info("<<<PASS %s: %s started on host %s at %s>>>\n", self.loop, name,
+                          reboot_host, time.ctime())
+            status = True
+            self.dmg_command.system_query()
+            # wait for node to complete rebooting
+            if not wait_for_result(self.log, check_ping, 60, 5, True, host=reboot_host,
+                                   expected_ping=True, cmd_timeout=60, verbose=True):
+                self.log.error(f"<<<FAILED: {name} - {reboot_host} failed to reboot")
                 status = False
-        if status:
-            # reintegrate ranks
-            reintegrate_status = True
-            for pool in pools:
-                for rank in ranklist:
+            if not wait_for_result(self.log, check_ssh, 120, 2, True, hosts=reboot_host,
+                                   cmd_timeout=30, verbose=True):
+                self.log.error(f"<<<FAILED: {name} - {reboot_host} failed to reboot")
+                status = False
+            if status:
+                # issue a restart
+                self.log.info("<<<PASS %s: Issue systemctl restart daos_server on %s at %s>>>\n",
+                              self.loop, name, reboot_host, time.ctime())
+                cmd_results = run_remote(
+                    self.log, reboot_host, "sudo systemctl restart daos_server")
```

Comment:
This function will use sudo args similar to elsewhere. E.g. `sudo -n`
```suggestion
                    self.log, reboot_host, command_as_user("systemctl restart daos_server", "root"))
```